R/init.R

```diff
@@ -222,27 +186,235 @@ init <- function(data,
     }
   )
 
+  if (length(landing) == 1L) {
+    res <- add_custom_server(res, function(input, output, session) {
+      do.call(landing[[1L]]$server, c(list(id = "landing_module_shiny_id")))
+    })
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "landing_popup_module()",
+      details = paste(
+        "landing_popup_module() is deprecated.",
+        "Use add_landing_popup() on the teal app object instead."
+      )
+    )
+  } else if (length(landing) > 1L) {
+    stop("Only one `landing_popup_module` can be used.")
+  }
+
   logger::log_debug("init teal app has been initialized.")
 
   res
 }
+
+#' Add a custom title to the app
+#'
+#' @param title The title to add
+#' @return The modified app object
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris, MTCARS = mtcars),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_title("Custom title")
+#'
+#' shinyApp(app$ui, app$server)
+modify_title <- function(
+    app,
+    title = "teal app",
+    favicon = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png") {
+  res <- app
+  res$ui <- function(request) {
+    title_tag <- tags$div(
+      id = "teal-title",
+      build_app_title(title, favicon)
+    )
+    ui_tq <- htmltools::tagQuery(app$ui(request = request))
+    ui_tq$find("#teal-title")$replaceWith(title_tag)$allTags()
+  }
+  res
+}
+
+#' Add a Header to a `teal` App
+#'
+#' @description Adds a header to the `teal` app.
+#'
+#' @param app (`environment`) The `teal` app environment.
+#' @param header (`shiny.tag` or `character(1)`) The header content to set. Defaults to an empty paragraph tag.
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_header(
+#'     tags$div(
+#'       h3("Custom header")
+#'     )
+#'   )
+#'
+#' shinyApp(app$ui, app$server)
+modify_header <- function(app, header = tags$p()) {
+  res <- app
+  res$ui <- function(request) {
+    ui_tq <- htmltools::tagQuery(app$ui(request = request))
+    ui_tq$find("#teal-header")$replaceWith(tags$header(id = "teal-header", header))$allTags()
+  }
+  res
+}
+
+#' Add a Footer to a `teal` App
+#'
+#' @description Adds a footer to the `teal` app.
+#'
+#' @param app (`environment`) The `teal` app environment.
+#' @param footer (`shiny.tag` or `character(1)`) The footer content to set. Defaults to an empty paragraph tag.
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_footer("Custom footer")
+#'
+#' shinyApp(app$ui, app$server)
+modify_footer <- function(app, footer = tags$p()) {
+  res <- app
+  res$ui <- function(request) {
+    ui_tq <- htmltools::tagQuery(app$ui(request = request))
+    ui_tq$find("#teal-footer")$replaceWith(tags$div(id = "teal-footer", footer))$allTags()
```

Comment:
```suggestion
    ui_tq$find("#teal-footer")$replaceWith(tags$footer(id = "teal-footer", footer))$allTags()
```

Reply:
We want to replace the div that encloses the footer in the `ui_teal`:
```r
tags$div(id = "teal-footer", footer)
```